server/src/main/java/io/unitycatalog/server/service/SchemaService.java

```diff
@@ -2,46 +2,161 @@
 import io.unitycatalog.server.persist.SchemaRepository;
+import io.unitycatalog.server.persist.model.Privileges;
+import io.unitycatalog.server.utils.IdentityUtils;
+import lombok.SneakyThrows;
+
+import java.util.List;
+import java.util.Map;
 import java.util.Optional;
+import java.util.UUID;
+
+import static io.unitycatalog.server.model.SecurableType.CATALOG;
+import static io.unitycatalog.server.model.SecurableType.METASTORE;
+import static io.unitycatalog.server.model.SecurableType.SCHEMA;
 
 @ExceptionHandler(GlobalExceptionHandler.class)
 public class SchemaService {
   private static final SchemaRepository SCHEMA_REPOSITORY = SchemaRepository.getInstance();
+  private static final CatalogRepository CATALOG_REPOSITORY = CatalogRepository.getInstance();
+  private final UnityCatalogAuthorizer authorizer;
+  private final UnityAccessEvaluator evaluator;
 
-  public SchemaService() {}
+  @SneakyThrows
+  public SchemaService(UnityCatalogAuthorizer authorizer) {
+    this.authorizer = authorizer;
+    evaluator = new UnityAccessEvaluator(authorizer);
+  }
 
   @Post("")
-  public HttpResponse createSchema(CreateSchema createSchema) {
-    return HttpResponse.ofJson(SCHEMA_REPOSITORY.createSchema(createSchema));
+  @AuthorizeExpression("""
+      #authorize(#principal, #metastore, OWNER) ||
+      #authorize(#principal, #catalog, OWNER) ||
+      #authorizeAll(#principal, #catalog, USE_CATALOG, CREATE_SCHEMA)
+      """)
+  @AuthorizeKey(METASTORE)
+  public HttpResponse createSchema(
+      @AuthorizeKey(value = CATALOG, key = "catalog_name") CreateSchema createSchema) {
+    SchemaInfo schemaInfo = SCHEMA_REPOSITORY.createSchema(createSchema);
+    createAuthorizations(schemaInfo);
+    return HttpResponse.ofJson(schemaInfo);
   }
 
   @Get("")
+  @AuthorizeExpression("#defer")
   public HttpResponse listSchemas(
       @Param("catalog_name") String catalogName,
       @Param("max_results") Optional<Integer> maxResults,
       @Param("page_token") Optional<String> pageToken) {
-    return HttpResponse.ofJson(SCHEMA_REPOSITORY.listSchemas(catalogName, maxResults, pageToken));
+    ListSchemasResponse listSchemasResponse =
+        SCHEMA_REPOSITORY.listSchemas(catalogName, maxResults, pageToken);
+    filterSchemas("""
+        #authorize(#principal, #metastore, OWNER) ||
+        (#authorize(#principal, #schema, OWNER) && #authorizeAll(#principal, #catalog, OWNER, USE_CATALOG)) ||
+        (#authorize(#principal, #schema, USE_SCHEMA) && #authorizeAll(#principal, #catalog, OWNER, USE_CATALOG))
+        """,
+        listSchemasResponse.getSchemas());
+    return HttpResponse.ofJson(listSchemasResponse);
   }
 
   @Get("/{full_name}")
-  public HttpResponse getSchema(@Param("full_name") String fullName) {
+  @AuthorizeExpression("""
+      #authorize(#principal, #metastore, OWNER) ||
+      (#authorizeAny(#principal, #schema, OWNER, USE_SCHEMA) && #authorizeAny(#principal, #catalog, OWNER, USE_CATALOG))
```

Comment:
Catalog owner and principals with USE_CATALOG and USE_SCHEMA (or owner) should be able to get